typescript/phoenix-sdk/src/utils/market.ts

```diff
@@ -628,3 +633,299 @@ export function getMarketExpectedOutAmount({
+    [
+      trader.toBuffer(),
+      TOKEN_PROGRAM_ID.toBuffer(),
+      marketData.header.baseParams.mintKey.toBuffer(),
+    ],
+    ASSOCIATED_TOKEN_PROGRAM_ID
+  )[0];
+  const quoteAccount = PublicKey.findProgramAddressSync(
+    [
+      trader.toBuffer(),
+      TOKEN_PROGRAM_ID.toBuffer(),
+      marketData.header.quoteParams.mintKey.toBuffer(),
+    ],
+    ASSOCIATED_TOKEN_PROGRAM_ID
+  )[0];
+
+  const limitOrderAccounts = {
+    phoenixProgram: PROGRAM_ID,
+    logAuthority,
+    market,
+    trader,
+    seat,
+    baseAccount,
+    quoteAccount,
+    baseVault: marketData.header.baseParams.vaultKey,
+    quoteVault: marketData.header.quoteParams.vaultKey,
+  }
+
+  let args: PlaceLimitOrderInstructionArgs = {
+    orderPacket,
+  }
+
+  return createPlaceLimitOrderInstruction(limitOrderAccounts, args)
+}
+
+/**
+ * Returns a limit order packet
+ * @param side The side of the order
+ * @param priceInTicks The price of the order in ticks
+ * @param numBaseLots The number of base lots to trade
+ * @param selfTradeBehavior The self trade behavior
+ * @param clientOrderId The client order id
+ * @param useOnlyDepositedFunds Whether to use only deposited funds
+ */
+export function getLimitOrderPacket(side: Side, priceInTicks: number, numBaseLots: number, selfTradeBehavior?: SelfTradeBehavior, matchLimit?: beet.COption<beet.bignum>, clientOrderId?: number, useOnlyDepositedFunds?: boolean): OrderPacket {
+  return {
+    __kind: "Limit",
+    side,
+    priceInTicks,
+    numBaseLots,
+    selfTradeBehavior: selfTradeBehavior ?? SelfTradeBehavior.CancelProvide,
+    matchLimit,
+    clientOrderId: clientOrderId ?? 0,
+    useOnlyDepositedFunds: useOnlyDepositedFunds ?? false,
+  }
+}
+/**
+ * Return all instructions necessary to place a limit order on a market, including
+ * - checking for and creating ATAs, if they do not exist
+ * - checking for and claiming a seat, if trader does not have a seat on the market
+ * - instruction for placing the limit order itself
+ * Useful if unknown whether trader has the correct ATAs and seat initialized 
+ * @param connection 
+ * @param market 
+ * @param marketData 
+ * @param trader 
+ * @param orderPacket 
+ * @returns 
+ */
+export async function getLimitOrderNewMakerIxs(connection: Connection, market: PublicKey, marketData: MarketData, trader: PublicKey, orderPacket: OrderPacket): Promise<TransactionInstruction[]> {
+  const instructions: TransactionInstruction[] = [];
+
+  let ataIxs = await getCreateTokenAccountInstructions(connection, trader, trader, marketData.header.baseParams.mintKey);
+  ataIxs.push(...await getCreateTokenAccountInstructions(connection, trader, trader, marketData.header.quoteParams.mintKey));
+  let claimSeatIx = await confirmOrCreateClaimSeatIxs(connection, market, trader);
+
+  instructions.push(...ataIxs);
+  instructions.push(...claimSeatIx);
+  instructions.push(getPlaceLimitOrderSingleIx(market, marketData, trader, orderPacket));
```

Comment:
why not use the existing getLimitOrder helper? or do we not have that?

Reply:
Yep, `getPlaceLimitOrderSingleIx` is supposed to function as the default `getPlaceLimitOrderIx`. We can rename to the latter if that's more clear